File: 0202-happy-number/0202-happy-number.cpp
class Solution {
public:
    bool isHappy(int n) {
        unordered_set<int> mySet;
        mySet.insert(n);
        while (true) {
            int sum = 0;
            while(n != 0) {
                sum += pow((n % 10), 2);
                n /= 10;
            }

            if (sum == 1) {
                return true;
            }

            n = sum;

            if (mySet.find(n) != mySet.end()) {
                return false;
            }

            mySet.insert(n);
        }

        return true;
    }
};

File: 0202-happy-number/0202-happy-number.py
class Solution:
    def isHappy(self, n: int) -> bool:

        numSet = set()

        while True:
            if n == 1:
                return True

            if n in numSet:
                return False
            
            numSet.add(n)

            

            n = sum(int(digit) ** 2 for digit in str(n))

        return False
